src/BlockTypesController.php

```diff
@@ -250,7 +250,7 @@ protected function get_block_types() {
 			);
 		}
 
-		return $block_types;
+		return apply_filters( 'woocommerce_blocks_block_types', $block_types );
```

Comment:
This seems like a good idea in practice, but want to check in with @mikejolley and @senadir to get their opinion here

Reply:
This would make our code more brittle. I'd prefer that if we did allow this, we'd group the blocks by type, e.g. Disable _all_ product blocks. Or disable _all_ cart/checkout blocks.

We don't want to run into issues where a block breaks because one of its other block dependencies is not registered, for example, the product block, which has its inner blocks registered separately.

Reply:
@mikejolley Maybe an array of dependencies would help? ie. if the dev wants to demove a block, that's dependency of another block and that block should still be registered, we can throw an exception. That way there's a clear link of the inner deps, and we do not loose flexibility.

Reply:
@vaclavgreif I think that could prove difficult to maintain, especially with things being so fast-moving.

Perhaps rather than this blunt tool to remove blocks, we could instead look at the reason why you'd want to remove certain block registrations? Is there a particular Block or part of the codebase that is causing you issues? Just product blocks?

Reply:
@mikejolley The only reason is performance - the block registration takes quite some time (0.1s on my test installation) and is by far the slowest part of default WooCommerce installation. As I do not use any of the blocks, I'd need a way to disable loading them.

Reply:
I wonder if it could be narrowed down to a particular block type. Registration is not a heavy operation, but some of the `initialize` calls may well be, and that could be something we can improve.

Reply:
If not, would it make sense to disable the whole blocks package?

Reply:
I did a quick test and drilled down to this: https://imgur.com/a/ZtfSNLB

Reply:
I've created an issue to track this https://github.com/woocommerce/woocommerce-blocks/issues/7979